### PR TITLE
Hongxin/test

### PR DESCRIPTION
**Pull request by glazhh**
add notebooks/predict.ipynb and notebooks/random_one_point_prediction.py
implement one random point prompt to sam and evaluate the performance by the mIoU and pixel acc

### CODE REVIEW

File: notebooks/random_one_point_prediction.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import sys
+sys.path.append("..")
+from segment_anything import sam_model_registry, SamAutomaticMaskGenerator, SamPredictor
+
+
+# numpy version
+def pixAcc(predicted, target):
+    same = (predicted == target).sum()
+    w, h = target.shape
+    return same / (w * h)
```
Comment (TakLee96) [inline notebooks/random_one_point_prediction.py:14]: nit:
```
return (predicted == target).mean()
```

File: notebooks/random_one_point_prediction.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        mask = np.all(mask_gt == label, axis=-1)
+        # 1 ramdon point from mask
+        num_point = 1
+        indices = np.argwhere(mask)
+        # swap x y
+        indices[:,[1,0]] = indices[:,[0,1]]
+        # sample on random point
+        point = np.asarray(indices[np.random.randint(indices.shape[0], size=num_point)])
+        # all point for test
+        points = np.asarray(indices)
```
Comment (TakLee96) [inline notebooks/random_one_point_prediction.py:41]: I think `np.argwhere` already returns a numpy array?

File: notebooks/random_one_point_prediction.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    img_miou_sum , img_pixacc_sum, num_class = 0, 0, len(anns)
+    for ann in anns:
+        target = ann['segmentation']
+        input_point = ann['random_point_1']
+        input_label = np.array([1])
+        masks, scores, logits = predictor.predict(
+            point_coords=input_point,
+            point_labels=input_label,
+            multimask_output=False,
+        )
```
Comment (TakLee96) [inline notebooks/random_one_point_prediction.py:88]: This version of code still runs predictor.predict for `20 x 10000` times, right?

File: notebooks/random_one_point_prediction.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        iou = IOU(masks[0],ann['segmentation'])
```
Comment (TakLee96) [inline notebooks/random_one_point_prediction.py:89]: masks[0] is assuming we're picking the highest confidence mask?
I imagine for the "oracle" version we can iterate over the top 3 masks and calculate the max IoU

File: notebooks/random_one_point_prediction.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    for label in labels:
+        # skip background
+        if label == 0:
+            continue
+        mask = np.all(mask_gt == label, axis=-1)
+        # 1 ramdon point from mask
+        num_point = 1
+        indices = np.argwhere(mask)
+        # swap x y
+        indices[:,[1,0]] = indices[:,[0,1]]
```
Comment (TakLee96) [inline notebooks/random_one_point_prediction.py:37]: This is because `np.argwhere` returns `i, j` indicies (which are effectively `y, x`) but SAM expects `x, y`?

File: notebooks/score.py
Comment (TakLee96) [inline notebooks/score.py:1]: This is only for our reference, right? We're not actually using the file?

**Review by TakLee96 — APPROVED** [review]
Looks good! ac with comments

File: notebooks/random_one_point_prediction.py (hunk 2 above)
Comment (glazhh) [reply notebooks/random_one_point_prediction.py:41]: Yep, I forgot that

File: notebooks/random_one_point_prediction.py (hunk 3 above)
Comment (glazhh) [reply notebooks/random_one_point_prediction.py:88]: Not every example have 20 class(and I also skip background,so It actually predict maximum 19 class) .  10000* num_class

File: notebooks/random_one_point_prediction.py (hunk 4 above)
Comment (glazhh) [reply notebooks/random_one_point_prediction.py:89]: I set multimask_output= False, so it only predict the highest score of top 3 masks. When I implement "oracle" after, I will try to sum the max mIoU of top 3 masks

File: notebooks/random_one_point_prediction.py (hunk 5 above)
Comment (glazhh) [reply notebooks/random_one_point_prediction.py:37]: Yes, I will leave comment in file later

File: notebooks/score.py
Comment (glazhh) [reply notebooks/score.py:1]: Yes, We're not actually using this file. it could be deleted
